### PR TITLE
Add isEmpty extension to remove on:content event from collaborator editor

### PR DESCRIPTION
**Pull request by annano**
# Contribution checklist

## Checklist

* [ ] - UI test added to added/changed functionality?
* [ ] - Screenshot is added to PR if applicable ?
* [ ] - Does the code work? Check whether function and logic are correct.
* [ ] - Does Changelog.md is updated with changes?
* [ ] - Does the translations are up to date?
* [ ] - Does it well tested?
* [ ] - Tested for Chrome.
* [ ] - Tested for Safari.
* [ ] - Go through the changed code looking for typos, TODOs, commented LOCs, debugging pieces of code, etc.
* [ ] - Rebase your branch onto master and upstream branch
* [ ] - Is there any redundant or duplicate code?
* [ ] - Are required links are linked to PR?
* [ ] - Does new code is well documented ?

## Related issues

A list of closed updated issues

### CODE REVIEW

File: packages/text-editor/src/components/extension/validateContent.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+}
+
+export interface ValidateContentStorage {
+  isEmpty: boolean
+}
+
+const handleChange = (editor: Editor, options: ValidateContentOptions, storage: ValidateContentStorage): void => {
+  if (options.onEmpty !== undefined && options.onEmpty !== null) {
+    if (storage.isEmpty !== editor.isEmpty) {
+      storage.isEmpty = editor.isEmpty
```
Comment (aonnikov) [inline packages/text-editor/src/components/extension/validateContent.ts]: Should we update storage even when not `onEmpty` callback was passed?

File: packages/text-editor/src/components/extension/validateContent.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      options.onEmpty(storage.isEmpty)
+    }
+  }
+
+  if (options.onValidate !== undefined && options.onValidate !== null) {
+    options.onValidate(editor.getHTML()) // TODO: might need to call async ?
```
Comment (aonnikov) [inline packages/text-editor/src/components/extension/validateContent.ts]: Yes, maybe async ... 
When I tried to edit large documents `getHTML` was slow. Maybe we should pass editor so the calling code will decide how to perform validation (check html, prosemirror json, or text).

File: packages/text-editor/src/components/extension/validateContent.ts (hunk 1 above)
Comment (lexiv0re) [inline packages/text-editor/src/components/extension/validateContent.ts]: Is `editor.isEmpty` = `true` for `<p></p>` value in the editor?

Comment (aonnikov) [reply packages/text-editor/src/components/extension/validateContent.ts]: It depends on the schema. It checks whether content matches the default content. If the content is block, then yes, `editor.isEmpty` will be `true` for `<p></p>`

https://github.com/ueberdosis/tiptap/issues/580#issuecomment-826398844
https://github.com/ueberdosis/tiptap/blob/b359f137e9777200e2483b1754ec1987e7d59765/packages/core/src/Editor.ts#L452

Comment (annano) [reply packages/text-editor/src/components/extension/validateContent.ts]: Yes

Comment (annano) [reply packages/text-editor/src/components/extension/validateContent.ts]: Updated

Comment (lexiv0re) [reply packages/text-editor/src/components/extension/validateContent.ts]: ok, great

File: packages/text-editor/src/components/extension/validateContent.ts (hunk 2 above)
Comment (annano) [reply packages/text-editor/src/components/extension/validateContent.ts]: I moved the extension for only `isEmpty` state, validation extension should probably be more complicated.

**Review by aonnikov — APPROVED** [review]

**Pull request merged**